src/modules/authentication/auth-components/RegisterForm.tsx

```diff
@@ -74,38 +60,41 @@ export const RegisterForm: React.FC = () => {
 
   return (
     <div className='reg-div'>
-      <form onSubmit={handleSignUp} className='reg-form'>
+      <form onSubmit={handleSubmit(onSubmit)} className='reg-form'>
         <h1>Create an account</h1>
         <div className='input-div'>
           <label htmlFor='email'>Email</label>
           <input
             type='email'
-            name='email'
+            {...register('email', { required: 'Email field is required.' })}
+            onChange={handleInputChange}
             placeholder='name@gmail.com'
-            onChange={handleEmailChange}
-            value={email}
-            required
+            id='email-register'
           />
+          {errors.email && <p>{errors.email.message}</p>}
           <label htmlFor='password'>Password</label>
           <input
             type={inputType}
-            name='password'
+            {...register('password', {
+              required: 'Password filed is required',
+              pattern: {
+                value:
+                  /^(?=.*[0-9])(?=.*[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?])[a-zA-Z0-9!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]{8,}$/g,
```

Comment:
bolje, ali mogao bi taj pattern izvuć u neku varijablu kao što sam ti dao u primjeru gore pa ga onda samo importat i koristit gdje ti treba. ovako je jako ružan render kod.